src/audio/up_down_mixer/up_down_mixer.c

```diff
@@ -401,44 +401,52 @@ static int up_down_mixer_init(struct processing_module *mod)
 
 static int
 up_down_mixer_process(struct processing_module *mod,
-		      struct input_stream_buffer *input_buffers, int num_input_buffers,
-		      struct output_stream_buffer *output_buffers, int num_output_buffers)
+		      struct sof_source **input_buffers, int num_input_buffers,
```

Comment:
please add sof_source sof_sink related header file into source header:
There should have: #include <sof/audio/sink_api.h> etc.

Reply:
There wasn't any audio_stream.h include so I don't see need adding sink_api.h

Reply:
this is not a reason, we can't make mistakes continue, previous did not add means we lose maintain on this(many reasons), now, we noticed this, we can change this step by step to make each source file correct.